rasa/core/policies/ted_policy.py

```diff
@@ -1697,10 +1748,21 @@ def batch_loss(
         tf_batch_data = self.batch_to_model_data_format(batch_in, self.data_signature)
         self._compute_dialogue_indices(tf_batch_data)
 
-        all_label_ids, all_labels_embed = self._create_all_labels_embed()
+        batch_label_ids = tf_batch_data[LABEL_KEY][LABEL_SUB_KEY][0]
 
-        label_ids = tf_batch_data[LABEL_KEY][LABEL_SUB_KEY][0]
-        labels_embed = self._get_labels_embed(label_ids, all_labels_embed)
+        (
+            sampled_label_ids,
+            sampled_labels_embed,
+        ) = self._create_all_labels_embed_for_training(
+            number_of_random_labels=self.config[LABEL_BATCH_SIZE]
+        )
+
+        batch_labels_embed = self._compute_embedding_for_label_ids(
```

Comment:
by doing separate computation, the number of computations are doubled compared to previous implementation